tests/tools/groups/test_paths.py

```diff
@@ -135,17 +136,17 @@ def test_walk_nodes(clear_database):
     assert [(r.group_path.path, r.node.attributes) for r in group_path.walk_nodes()] == [('a', {'i': 1, 'j': 2})]
 
 
-def test_type_string(clear_database_before_test):
-    """Test that only the type_string instantiated in ``GroupPath`` is returned."""
+def test_cls(clear_database_before_test):
+    """Test that only instances of `cls` or its subclasses are matched by ``GroupPath``."""
     for label in ['a', 'a/b', 'a/c/d', 'a/c/e/g']:
-        orm.Group.objects.get_or_create(label, type_string=orm.GroupTypeString.USER.value)
+        orm.Group.objects.get_or_create(label)
     for label in ['a/c/e', 'a/f']:
-        orm.Group.objects.get_or_create(label, type_string=orm.GroupTypeString.UPFGROUP_TYPE.value)
+        orm.UpfFamily.objects.get_or_create(label)
     group_path = GroupPath()
     assert sorted([c.path for c in group_path.walk()]) == ['a', 'a/b', 'a/c', 'a/c/d', 'a/c/e', 'a/c/e/g']
-    group_path = GroupPath(type_string=orm.GroupTypeString.UPFGROUP_TYPE.value)
+    group_path = GroupPath(cls=orm.UpfFamily)
     assert sorted([c.path for c in group_path.walk()]) == ['a', 'a/c', 'a/c/e', 'a/f']
-    assert GroupPath('a/b/c') != GroupPath('a/b/c', type_string=orm.GroupTypeString.UPFGROUP_TYPE.value)
+    assert GroupPath('a/b/c') != GroupPath('a/b/c', cls=orm.UpfFamily)
 
 
```

Comment:
```suggestion

def test_cls_label_clashes(clear_database_before_test):
    """Test behaviour when multiple group classes have the same label."""

    grp1, _ = orm.Group.objects.get_or_create("a")
    node1 = orm.Data()
    node1.store()
    grp1.add_nodes(node1)

    grp2, _ = orm.UpfFamily.objects.get_or_create("a")
    node2 = orm.Data()
    node2.store()
    grp2.add_nodes(node2)

    assert GroupPath("a").group_ids == [grp1.id]
    with pytest.raises(MultipleObjectsError):
        GroupPath("a").get_group()
    assert [(r.group_path.path, r.node.id) for r in GroupPath("a").walk_nodes()] == [("a", node1.id)]

    assert GroupPath("a", cls=orm.UpfFamily).group_ids == [grp2.id]
    assert GroupPath("a", cls=orm.UpfFamily).get_group().id == grp2.id
    assert [(r.group_path.path, r.node.id) for r in GroupPath("a", cls=orm.UpfFamily).walk_nodes()] == [("a", node2.id)]
```

This is the main thing I wanted to check. I'm not sure if it is desired behaviour that `GroupPath("a").get_group()` raises an exception. Should it act the same as `group_ids`, such that it does not allow sub-classing. Also should there be a `get_groups` that does allow sub-classing?